src/libostree/ostree-repo.c

```diff
@@ -4171,6 +4176,41 @@ repo_load_file_archive (OstreeRepo *self,
     }
 }
 
+static GVariant *
+_ostree_repo_read_detached_xattrs (OstreeRepo   *self,
+                                   const char   *checksum,
+                                   GCancellable *cancellable,
+                                   GError      **error)
+{
+  g_assert (self != NULL);
+
+  g_autofree char *xattr_path = NULL;
+  xattr_path = g_strdup_printf ("objects/%c%c/%s.file.xattrs",
```

Comment:
Two minor nits; why not do declare-and-initialize i.e.:
`g_autofree char *xattr_path = g_strdup_printf (...)`?

But second, we could also consider a stack allocation here.  At one point I was trying to reduce the number of mallocs we do in potentially hot paths of reading lots of files.  I don't have a strong opinion on that though.  We can come back to it later.

Reply:
Agreed. I do plan to eventually fold this back into the existing `_ostree_loose_path()` logic, which does not involve the heap.

However I first wanted to ask whether the double `.file.xattrs` extension here happened on purpose or by mistake?
It is the only object-type with a double suffix, and I can foresee that may cause future headaches.

It gets suffixed [here in ostree-rs-ext](https://github.com/ostreedev/ostree-rs-ext/blob/d1466efcb949e1359d377e8d3fca11f3913d61e4/lib/src/tar/export.rs#L306), which made me wonder whether it was actually intentional.
I'd be inclined to turn that into a single `.file-xattrs` extension (or similar) instead, as part of the v1 format.

Reply:
Single extension makes sense to me.

Reply:
We can do that change in format version 1